### PR TITLE
Exposing twilio warning

### PR DESCRIPTION
**Pull request by charliesantos**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

## Pull Request Details

Exposing TwilioWarning that way developers can also do 

```js
import { connect, TwilioWarning } from 'twilio-video';

const room = await connect('token', {
  notifyWarnings: [TwilioWarning.recordingMediaLost]
  // Other connect options
});
```

### Before review
* [ ] Updated CHANGELOG.md if necessary
* [ ] Added unit tests if necessary
* [ ] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [ ] Ready for review

### CODE REVIEW

**Comment by makarandp0** [comment]
@charliesantos, It feels bit odd that we have `TwiloError` which is a class, but `TwilioWarning` is an enum. 
Why not recommend that they connect with a string value instead of exposing this enum?
```
const room = await connect('token', {
  notifyWarnings: ["recording-media-lost"]
  // Other connect options
});
```

**Comment by charliesantos** [comment]
> @charliesantos, It feels bit odd that we have `TwiloError` which is a class, but `TwilioWarning` is an enum. Why not recommend that they connect with a string value instead of exposing this enum?
> 
> ```
> const room = await connect('token', {
>   notifyWarnings: ["recording-media-lost"]
>   // Other connect options
> });
> ```

I was thinking to do it this way to make it more strict on TS environment. But I understand your point. Let me adjust it for TS.

**Comment by charliesantos** [comment]
@makarandp0 can you please take another look?

File: tsdef/index.d.ts [hunk 1]
```diff
@@ -81,6 +80,7 @@ export {
   StatsReport,
   TrackStats,
   TrackSwitchOffMode,
+  TwilioWarning,
```
Comment (makarandp0) [inline tsdef/index.d.ts:83]: is this needed anymore?

Comment (charliesantos) [reply tsdef/index.d.ts:83]: Yes for TS. That way we can do this

```ts
videoPublication?.on('warning', (name: TwilioWarning) => {

});
```

**Comment by makarandp0** [comment]
LGTM!

**Pull request merged**